pkg/model/components/kubelet.go

```diff
@@ -81,11 +81,15 @@ func (b *KubeletOptionsBuilder) BuildOptions(o interface{}) error {
 	}
 
 	if clusterSpec.Kubelet.ClusterDNS == "" {
-		ip, err := WellKnownServiceIP(clusterSpec, 10)
-		if err != nil {
-			return err
+		if clusterSpec.KubeProxy.ProxyMode == "ipvs" && !clusterSpec.KubeDNS.NodeLocalDNS.Disable {
+			clusterSpec.Kubelet.ClusterDNS = "169.254.20.10"
```

Comment:
This is also a configurable field. It is better to add validation logic stating that you need to set this IP to a certain value if you enable this addon.
There is already some validation logic around this in https://github.com/kubernetes/kops/blob/b0cb7c1795a76bb1bd5d44542bdb11dae12ec92f/pkg/apis/kops/validation/legacy.go

Reply:
Yeah make sense. I'll make wire some changes to take into account validation